SU2_CFD/src/solvers/CTurbSASolver.cpp

```diff
@@ -1595,123 +1594,59 @@ void CTurbSASolver::SetNuTilde_WF(CGeometry *geometry, CSolver **solver_containe
-
-    /*--- Now compute the Eddy viscosity at the first point off of the wall ---*/
-
-    su2double Lam_Visc_Normal = flow_nodes->GetLaminarViscosity(Point_Normal);
-    su2double Density_Normal = flow_nodes->GetDensity(Point_Normal);
+    su2double Lam_Visc_Normal = flow_nodes->GetLaminarViscosity(iPoint_Neighbor);
+    su2double Density_Normal = flow_nodes->GetDensity(iPoint_Neighbor);
     su2double Kin_Visc_Normal = Lam_Visc_Normal/Density_Normal;
 
-    su2double dypw_dyp = 2.0*Y_Plus_White*(kappa*sqrt(Gam)/Q)*sqrt(1.0 - pow(2.0*Gam*U_Plus - Beta,2.0)/(Q*Q));
-    su2double Eddy_Visc = Lam_Visc_Wall*(1.0 + dypw_dyp - kappa*exp(-1.0*kappa*B)*
-                                         (1.0 + kappa*U_Plus + kappa*kappa*U_Plus*U_Plus/2.0)
-                                         - Lam_Visc_Normal/Lam_Visc_Wall);
-
-    /*--- Eddy viscosity should be always a positive number ---*/
-
-    Eddy_Visc = max(0.0, Eddy_Visc);
+    su2double Eddy_Visc = solver_container[FLOW_SOL]->GetEddyViscWall(val_marker, iVertex);
 
     /*--- Solve for the new value of nu_tilde given the eddy viscosity and using a Newton method ---*/
 
+    // start with positive value of nu_til_old
+    su2double nu_til = 0.0; 
     su2double nu_til_old = nodes->GetSolution(iPoint,0);
 
     unsigned short counter = 0;
     su2double diff = 1.0;
 
     while (diff > tol) {
+      // note the error in Nichols and Nelson 
+      su2double func = nu_til_old*nu_til_old*nu_til_old*nu_til_old - (Eddy_Visc/Density_Normal)*(nu_til_old*nu_til_old*nu_til_old + Kin_Visc_Normal*Kin_Visc_Normal*Kin_Visc_Normal*cv1_3);
+      su2double func_prim = 4.0 * nu_til_old*nu_til_old*nu_til_old - 3.0*(Eddy_Visc/Density_Normal)*(nu_til_old*nu_til_old);
 
-      su2double const_term = (Eddy_Visc/Density_Normal) * pow(Kin_Visc_Normal,3)*cv1_3;
-
-      su2double nu_til_2 = pow(nu_til_old,2);
-      su2double nu_til_3 = nu_til_old * nu_til_2;
-
-      su2double func = nu_til_old * (nu_til_3 - (Eddy_Visc/Density_Normal)*nu_til_2) + const_term;
-      su2double func_prime = 4.0*nu_til_3 - 3.0*(Eddy_Visc/Density_Normal)*nu_til_2;
-      su2double nu_til = nu_til_old - func/func_prime;
+      // damped Newton method
+      nu_til = nu_til_old - 0.5*(func/func_prim);
 
       diff = fabs(nu_til-nu_til_old);
       nu_til_old = nu_til;
 
+      // sometimes we get negative values when the solution has not converged yet, we just reset the nu_tilde in that case.
+      if (nu_til_old<tol) {
+        //cout <<"nu_tilde =0, resetting!"<<endl;
+        nu_til_old = 0.001;
+        nu_til=0.002;
+      }
+
       counter++;
       if (counter > max_iter) {
         cout << "WARNING: Nu_tilde evaluation has not converged." << endl;
         break;
       }
     }
 
-    nodes->SetSolution_Old(Point_Normal, 0, nu_til_old);
-    LinSysRes.SetBlock_Zero(Point_Normal);
+    su2double Sol[nVar];
+    for (auto iVar = 0u; iVar < nVar; iVar++)
+      Sol[iVar] = nu_til;
```

Comment:
This dynamic c-style array is not standard c++, you should use MAXNVAR which is a compile time constant, making the array static, or in the case of SA which we know to have only 1 variable... no arrays are needed.

Reply:
changed nvar to MAXNVAR.